pkg/conduit/runtime.go

```diff
@@ -461,7 +469,38 @@ func (r *Runtime) serveGRPCAPI(ctx context.Context, t *tomb.Tomb) (net.Addr, err
 	grpc_health_v1.RegisterHealthServer(grpcServer, healthServer)
 
 	// serve grpc server
-	return r.serveGRPC(ctx, t, grpcServer)
+	return r.serveGRPC(ctx, t, grpcServer, r.Config.API.GRPC.Address)
+}
+
+// startConnectorUtils starts all the utility services neededed by connectors.
+func (r *Runtime) startConnectorUtils(ctx context.Context, t *tomb.Tomb) (net.Addr, error) {
+	grpcServer := grpc.NewServer(
+		grpc.ChainUnaryInterceptor(
+			grpcutil.RequestIDUnaryServerInterceptor(r.logger),
+			grpcutil.LoggerUnaryServerInterceptor(r.logger),
+		),
+		grpc.StatsHandler(r.gRPCStatsHandler),
+	)
+
+	schemaServiceAPI := utils.NewSchemaServiceAPIv1(r.schemaService)
+	schemaServiceAPI.RegisterInServer(grpcServer)
+
+	// Makes it easier to use command line tools to interact
+	// with the gRPC API.
+	// https://github.com/grpc/grpc/blob/master/doc/server-reflection.md
+	reflection.Register(grpcServer)
+
+	// Names taken from schema.proto
+	healthServer := api.NewHealthServer(
+		map[string]api.Checker{
+			"SchemaService": r.schemaService,
+		},
+		r.logger,
+	)
+	grpc_health_v1.RegisterHealthServer(grpcServer, healthServer)
+
+	// todo make port random
+	return r.serveGRPC(ctx, t, grpcServer, ":8184")
```

Comment:
Hint: `":0"` will choose a random free port. Or if we want to give the user some control over this, we could do the same as go-plugin and let the user define [min and max ports](https://github.com/hashicorp/go-plugin/blob/721087dad1e61cd41a19a30e5ae740dceb55061d/client.go#L188-L191).

Reply:
Yup, yup, the port stuff is going to be handled in #1563.